src/KUSignalLib/DSP/DSP.py

```diff
@@ -25,4 +26,114 @@ def IIRDirectForm2(b, a, x):
         tmp = np.dot(denominator, delayLine) #df2 left side
         delayLine[1:] = delayLine[:-1] #shift delay line
         delayLine[0] = x[i]*a[0] + tmp #new value is x[n] * a[0] + sum of left side
-    return y
\ No newline at end of file
+    return y
+
+def Interpolate(x, n, mode="linear"):
```

Comment:
if this is just being used as a helper function, lets make it private(not sure how to do this in python.........)